packages/maskbook/src/utils/native-rpc/types.ts

```diff
@@ -0,0 +1,27 @@
+import type * as settings from '../../settings/settings'
+/**
+ * JSON RPC calls that can be called from the native side.
+ * JS = server, native = client
+ */
+export interface WebviewAPIs {
+    web_echo<T>(arg: T): Promise<T>
+    getDashboardUrl(path: string): Promise<string>
+    getSettings(key: keyof typeof settings): Promise<any>
```

Comment:
There are still hidden dependencies on the internal implementation. The other side is not TypeScript and does not have direct dependencies on our codebase. Please explicitly list the settings you want to expose and mark their types.

Reply:
I suggest it should be like:

```ts
getSettings(key: "settings-name-A"): Promise<SettingsReturnValueA>
getSettings(key: "settings-name-B"): Promise<SettingsReturnValueB>
```

use overload to mark what settings can be used and it will easier to read by @jk234ert

Reply:
ok